### PR TITLE
Add indexes for expression, query, and error docs.

### PR DESCRIPTION
**Pull request by sigmod**

### CODE REVIEW

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

File: content/analytics/3_query.html [hunk 1]
```diff
@@ -8,9 +8,61 @@
...
+  <li><a href="#Introduction">Introduction</a></li>
+  <li><a href="#SELECT_statements">SELECT Statements</a></li>
+  <li><a href="#Select_clauses">SELECT Clauses</a>
+    <ul>
+      <li><a href="#Select_element">Select Element/Value/Raw</a></li>
+      <li><a href="#SQL_select">SQL-style Select</a></li>
+      <li><a href="#Select_star">Select *</a></li>
+      <li><a href="#Select_distinct">Select Distinct</a></li>
+      <li><a href="#Unnamed_projections">Unnamed Projections</a></li>
+      <li><a href="#Abbreviatory_field_access_expressions">Abbreviatory Field Access Expressions</a></li>
```
Comment (westmann) [inline content/analytics/3_query.html]: Abbreviated?

**Comment by westmann** [comment]
@cb-docs-robot build analytics

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

**Comment by sigmod** [comment]
@cb-docs-robot build analytics

**Comment by mattcarabine** [comment]
@sigmod do you know where the bullet list at the beginning has come from?

**Comment by sigmod** [comment]
@mattcarabine they come from markdown files, for example:

---
title: Expressions
description: A description of Couchbase Analytics SQL++ expressions.
layout: default
---

* [Introduction](#Introduction)
* [Operator Expressions](#Operator_expressions)
     * [Arithmetic Operators](#Arithmetic_operators)
     * [Collection Operators](#Collection_operators)
     * [Comparison Operators](#Comparison_operators)
     * [Logical Operators](#Logical_operators)
* [Case Expressions](#Case_expressions)
* [Quantified Expressions](#Quantified_expressions)
* [Path Expressions](#Path_expressions)
* [Primary Expressions](#Primary_expressions)
     * [Literals](#Literals)
     * [Variable References](#Variable_references)
     * [Parenthesized Expressions](#Parenthesized_expressions)
     * [Function call Expressions](#Function_call_expressions)
     * [Constructors](#Constructors)

**Comment by mattcarabine** [comment]
@cb-docs-robot build analytics

**Comment by mattcarabine** [comment]
@sigmod Is that removable? Not sure it fits in with the general aesthetic of the page.

**Comment by sigmod** [comment]
@mattcarabine   the issue is that the three HTMLs are too large and needs an easy way for locating a certain topic.  Is there a way to automatically generate an index and put it somewhere on the page?

**Comment by mattcarabine** [comment]
@sigmod I can look at adding that, yes

Similarly to those on the main site: https://developer.couchbase.com/documentation/server/4.6/introduction/intro.html

**Comment by sigmod** [comment]
@mattcarabine the main site is fantastic.  It'll be great if we can get the index during html generation.

**Comment by mattcarabine** [comment]
Ok I'll look at doing that. In the meantime could you remove the links in your HTML (since I will be generating them instead)?

**Comment by sigmod** [comment]
@mattcarabine Sure, I'll close this PR.

**Pull request closed without merging**